doc/start/kube-helm.rst

```diff
@@ -0,0 +1,344 @@
+================================
+Installation (Kubernetes + Helm)
+================================
+
+The ceph-helm_ project enables you to deploy Ceph in a Kubernetes environement.
+This documentation assumes a Kubernetes environement is available.
+
+Current limitations
+===================
+
+ - The public and cluster networks must be the same
+ - If the storage class user id is not admin, you will have to manually create the user
+   in your Ceph cluster and create its secret in Kubernetes
+ - ceph-mgr can only run with 1 replica
+
+Install and start helm
+======================
+
+Helm can be installed by following these instructions_.
+
+Once installed, run::
+
+    $ helm init
+    $ helm serve &
+    $ helm repo add local http://localhost:8879/charts
+
+Add ceph-helm to Helm local repos
+==================================
+::
+
+    $ git clone https://github.com/ceph/ceph-helm
+    $ cd ceph-helm/ceph
+    $ make
+
+Configure your Ceph cluster
+===========================
+
+Create a ``~/ceph-overrides.yaml`` that will contain your Ceph configuration::
+
+    $ cat ~/ceph-overrides.yaml
+    network:
+      public:   172.21.0.0/20
+      cluster:   172.21.0.0/20
+
+    osd_devices:
+      - name: dev-sdd
+        device: /dev/sdd
+        zap: "1"
+      - name: dev-sde
+        device: /dev/sde
+        zap: "1"
+
+    storageclass:
+      name: ceph-rbd
+      pool: rbd
+      user_id: k8s
+
+.. note:: If journal is not set it will be colocated with device
+
+.. note:: The ``ceph-helm/ceph/ceph/values.yaml`` file contains the full
+          list of option that can be set
+
+Configure RBAC permissions
+==========================
+
+Kubernetes >=v1.6 makes RBAC the default admission controller. ceph-helm provides RBAC roles and permissions for each component::
+
+    $ kubectl create -f ~/ceph-helm/ceph/rbac.yaml
+
+The ``rbac.yaml`` file assumes that the Ceph cluster will be deployed in the ``ceph`` namespace.
+
+Label kubelets
+==============
+
+The following labels need to be set to deploy a Ceph cluster:
+ - ceph-mon=enabled
+ - ceph-mgr=enabled
+ - ceph-osd=enabled
+ - ceph-osd-device-<name>=enabled
+
+The ``ceph-osd-device-<name>`` label is created based on the osd_devices name value defined in our ``ceph-overrides.yaml``.
+From our example above we will have the two following label: ``ceph-osd-device-dev-sdb`` and ``ceph-osd-device-dev-sdc``.
+
+For each Ceph Monitor::
+
+    $ kubectl label node <nodename> ceph-mon=enabled ceph-mgr=enabled
+
+For each OSD node::
+
+    $ kubectl label node <nodename> ceph-osd=enabled ceph-osd-device-dev-sdb=enabled ceph-osd-device-dev-sdc=enabled
+
+Ceph Deployment
+===============
+
+Run the helm install command to deploy Ceph::
+
+    $ helm install --name=ceph local/ceph --namespace=ceph -f ~/ceph-overrides.yaml
```

Comment:
Can we also specify how to use stable versions, if its coming from namespace?

Reply:
namespace doesn't have anything to do with version; it's a kubernetes segregation mechanism.  install images are selected by 'images:' in values.yaml (and overridden at helm install time with -e or -f ceph-overrides.yaml)

Reply:
Thanks @dmick